hack/build/build-test.sh

```diff
@@ -17,19 +17,37 @@
 # Copyright 2017 Red Hat, Inc.
 #
 
-set -e
+set -eo pipefail
 
 source hack/build/common.sh
 source hack/build/config.sh
 
+target="${1}"
+if ! [[ ${target} =~ (\.\/).*(\.\.\.)$ ]]; then
+     echo "Error: \"$1\" does not match expected go pattern (e.g. ./path/to/package/...)"
+     exit 1
+fi
+shift 1
+
+while true; do
+    case "${1}" in
+        --test-args=*)
+            test_args="${1##*=}"
+            echo "test-args: $test_args"
+            shift 1
+            ;;
+        *)
+            break
+            ;;
+    esac
+done
+
 kubectl=${CDI_DIR}/cluster/.kubectl
 
-if [[ ${TARGET} == openshift* ]]; then
+if [[ ${CLIENT} == openshift ]]; then
     oc=${kubectl}
 fi
 
-echo ${kubectl}
+echo "kubectl: $kubectl"
 
-kubeconfig=${CDI_DIR}/cluster/.kubeconfig
-# ${TESTS_OUT_DIR}/tests.test -kubeconfig=${kubeconfig} -oc-path=${oc} -kubectl-path=${kubectl} -test.timeout 90m ${FUNC_TEST_ARGS}
-go test -v ./tests/... -args -ginkgo.v -kubeconfig=${kubeconfig} -oc-path=${oc} -kubectl-path=${kubectl} -test.timeout 90m ${FUNC_TEST_ARGS}
+go test -v -test.timeout 90m ${target} -args ${test_args}
```

Comment:
We will need to converge this with #352 so all those environment variables are passed properly.